org.uqbar.project.wollok/src/org/uqbar/project/wollok/scoping/cache/WollokGlobalScopeCache.xtend

```diff
@@ -0,0 +1,11 @@
+package org.uqbar.project.wollok.scoping.cache
+
+import java.util.List
+import org.eclipse.emf.common.util.URI
+import org.eclipse.xtext.xbase.lib.Functions.Function0
+import org.uqbar.project.wollok.wollokDsl.Import
+import org.eclipse.xtext.resource.IEObjectDescription
+
+interface WollokGlobalScopeCache {
```

Comment:
There is two caches:

- One for the location of the "root file", the key here is the string representation of the path of the containing directory, so if there are many files in the same folder it will resolve only once. If you move the .wlk file to other directory it will recalculate for that directory. If you move the "root file" everything is doom. That is why I have put it hidden. I rather prefer not to be visible. 

- The other cache is for the imported objects, this cache use as key the string representation of the URI. In the cache I'm storing an object having the URI, the imports and the EObjectDescriptions. So before returning the value the list of imports is checked. If the list of imports changes the imported references are reloaded.

Now I am thinking if you change the imported file in the other world, it maybe is not updated. Maybe I should invalidate all the cache when a file is saved. I will do that.